dask_bigquery/tests/test_core.py

```diff
@@ -19,6 +20,7 @@ def df():
         {
             "name": random.choice(["fred", "wilma", "barney", "betty"]),
             "number": random.randint(0, 100),
+            "timestamp": datetime.now(timezone.utc) - timedelta(days=i % 2),
```

Comment:
Before adding `timezone.utc` I was getting this assertion error:
```python
E           AssertionError: Attributes of DataFrame.iloc[:, 2] (column name="timestamp") are different
E           
E           Attribute "dtype" are different
E           [left]:  datetime64[ns, UTC]
E           [right]: datetime64[ns]
```
It seems like when reading back from bigquery, it will automatically convert to utc if not otherwise specified, causing the error. 
@tswast can you confirm this is the case? any comments?

Reply:
`TIMESTAMP` columns are intended to come back as `datetime64[ns, UTC]`, yes.

`DATETIME` should come back as `datetime64[ns]`.

See my answer here on the difference between the two: https://stackoverflow.com/a/47724366/101923

Also note: both will come back as `object` dtype if there's a date outside of the pandas representable range, e.g. 0001-01-01 or 9999-12-31.

Reply:
I'm actually working on making the pandas-gbq dtypes consistent with google-cloud-bigquery as we speak in https://github.com/googleapis/python-bigquery-pandas/pull/444

Reply:
It seems that if I don't provide a schema, bigquery will infer that the dataframe column named `"timestamp"` is a `TIMESTAMP` column therefore it's converting it is coming back as `datetime64[ns, UTC]`. That been said to keep the test simple I think we can have the local dataframe to be timezone aware and test that it comes back as it should. 

cc: @jrbourbeau Does this convince you? If so this PR is ready for review.

Reply:
Sounds good 👍